### PR TITLE
Server components list, create, update support

### PR DESCRIPTION
**Pull request by joelrebel**
This PR extends the server components endpoint to support various operations
   
- Adds API endpoints to handle `GET`, `POST`, `PUT` queries on `ServerComponents`.
- Implements API endpoint handler methods to List, Get, Create and Update `ServerComponent` objects.
- Extends the ClientInterface, and implements methods to List, Get, Create and Update `ServerComponent`.
- Updates the ServerComponentType ID to be the UUID instead of the slug.
- A few fixes and adds tests for query parameter to attribute parsing.

```
    List server components:
    --
    GET /servers/components
    
     The List endpoint lists ServerComponents by the the query parameter,
    regardless of the server its referenced by - this is to be used to look
    up components that share a serial, model, vendor, attribute or versioned attribute.
    
    List server components referenced by the server UUID:
    --
    Get /servers/:uuid/component
    
    The Get endpoint lists ServerComponents limited to the given server UUID URL parameter.
    
    Create one or more server components
    --
    POST /servers/:uuid/component
    
    The Create endpoint creates one or more Component object(s), a client is expected to
    query the current list of components and explicitly invoke the Create
    endpoint when a component was not found in the results returned by the server.
    
    Update one or more server components
    --
    POST /servers/:uuid/component
    
    The Update endpoint updates one or more Component objects(s), a client
    is expected to know the component ID before hand by querying existing
    component information and merging in any component attribute changes
    identified on the hardware when invoking this endpoint.
 ```

### CODE REVIEW

File: pkg/api/v1/router_server_components.go [hunk 1]
```diff
@@ -2,18 +2,71 @@ package serverservice
...
 import (
 	"github.com/gin-gonic/gin"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
 	"github.com/volatiletech/sqlboiler/v4/queries/qm"
 )
 
+// serverComponentList returns a response with the list of components that matched the params.
 func (r *Router) serverComponentList(c *gin.Context) {
+	pager := parsePagination(c)
```
Comment (JAORMX) [inline pkg/api/v1/router_server_components.go:12]: similar to what @diogomatsubara did in https://github.com/metal-toolbox/hollow-serverservice/pull/48/files , seems to me like we would want pagination to already happen [on the SQL query](https://github.com/metal-toolbox/hollow-serverservice/pull/48/files) , that way, we would reduce the memory footprint from this server and be more resilient to DDoS's from this side.

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: pkg/api/v1/router_server_components.go (hunk 1 above)
Comment (joelrebel) [reply pkg/api/v1/router_server_components.go:12]: @JAORMX ack, let me see how that can be fixed. Which SQL query are you referring to in your comment, the link leads me to #48

Comment (JAORMX) [reply pkg/api/v1/router_server_components.go:12]: @joelrebel copypasta error, here it is: https://github.com/metal-toolbox/hollow-serverservice/blob/6297dc0251f020f85a9576c700ad72bf53b683bc/pkg/api/v1/router_firmware.go#L31

Comment (joelrebel) [reply pkg/api/v1/router_server_components.go:12]: @JAORMX we currently have a [default limit](https://github.com/metal-toolbox/hollow-serverservice/blob/ec5311e83f5978442769aed93c7f5a0cab8a1daa/pkg/api/v1/pagination.go#L14) of 100 items, it does seem to be on the higher side though. Are you suggesting pagination be done within the default limit?


```
SELECT "server_components".* FROM "server_components" LEFT JOIN versioned_attributes as 
... <snip> ...
AND json_extract_path_text(server_components_ver_attr_0.data::JSONB, $4) = $5)) ORDER BY server_components.created_at DESC LIMIT 100;    
```

**Review by JAORMX — APPROVED** [review]

**Pull request merged**